p2p/proto/requests.proto

```diff
@@ -0,0 +1,6 @@
+syntax = "proto3";
+
+import "p2p/proto/common.proto";
+import "p2p/proto/propagation.proto";
+
+
```

Comment:
we need a generic message that can carry different requests.

something like;

```message Request {
  oneof req {
    GetBlocks get_blocks = 1;
    GetSignatures get_signatures = 2;
    GetEvents get_events = 4;
    GetReceipts get_receipts = 5;
    GetTransactions get_transactions = 6;
  }
}
```

Reply:
I think the problem is, its unclear how to use these message with libp2p. One example is this:
https://github.com/starknet-io/starknet-p2p-specs/pull/7/files#diff-2d12e4bd8f0f7ff26ac5b3b4bd3f7b77e7bf68f02f46304f0c29b86716c56d04R53

But this does not differentiate the protocol between different request-response and just unify them under one message.

You might wanna make it clear which message request-response uses which protocol string, 

for example, unify all request-response into `/starknet/block-sync/1` protocol id like the comment, 

or have different protocol id for each message.
`/starknet/snap-sync/1/get-header` for just the get headers,
`/starknet/snap-sync/1/get-receipts` just the receipts
`/starknet/snap-sync/1/get-bodies`. and so on...

Reply:
> I thought we agreed to use different protocols

I must have missed that. So each file is describing a separate protocol, am I understanding correctly? or is it each `Get*` message?

Reply:
Yes, I agree I should note the which messages go into what protocols. Will add